File: leetcode/1easy/위성원/1897.ts
function makeEqual(words: string[]): boolean {
  if (words.length === 1) {
    return true
  }

  const len = words.reduce((prev, curr) => {
    return prev + curr.length
  }, 0)

  if (len % words.length !== 0) {
    return false
  }

  const al = new Array(26).fill(0);
  for (const word of words) {
    for (const c of word) {
      al[c.charCodeAt(0) - 97]++
    }
  }

  const good = al.every((n) => n % words.length === 0)

  return good
};

File: leetcode/1easy/이경호/1897.cpp
class Solution {
public:
  bool makeEqual(vector<string>& words) {
    map<char, int> cMap;
    for (auto word : words) {
      for (char c : word) {
        cMap[c]++;
      }
    }
    for (auto& [key, value] : cMap) {
      if (value % words.size()) {
        return false;
      }
    }
    return true;
  }
};